docs_src/design/ucr/0004-Provision-Watch-more-Device-Properties.md

```diff
@@ -0,0 +1,78 @@
+## Provision Watch more Device Properties
+This UCR describes the Use Case for Provision Watching via Additional Device Properties,
+beyond the protocol properties currently used exclusively for matching in Provision Watchers.
+
+### Submitters
+- Tom Brennan (Eaton)
+
+## Change Log
+- [pending](https://github.com/edgexfoundry/edgex-docs/pulls) (2022-07-26)
+
+
+### Market Segments
+Any that deploy EdgeX systems with analytic, utility, or north-bound microservices that must "discover"
+Devices added to the EdgeX core-metadata by south-bound Device Services.
+
+### Motivation
+The autodiscovery of Devices using Provision Watchers is a useful feature of Device Services; currently,
+the Provision Watcher implementation in the two Device SDKs uses the protocol properties of a discovered
+Device to match against the "identifiers" specified in the Provision Watcher metadata. The 
+implementations use regular expression matching against the "identifiers", and also filter out any 
+Devices whose protocol properties match the "blockingIdentifiers" of the Provision Watcher metadata.
+
+We are finding that [Hybrid App-Device Services](https://github.com/edgexfoundry/edgex-docs/pulls) *later (./0005-Hybrid-App-Device-Services.md)* also want to 
+use Provision Watchers, so that they can be configured at run-time to work with new Devices, but these 
+do not need or want to match the protocol properties of a Device; instead, they want to match or exclude
+based on Device properties such as the "profileName" (as a proxy for Device Model Name), "name", and "labels".
```

Comment:
First thought here is that `profileName` isn't assigned to a device until it matches a Provision Watcher, which has the  `profileName` to be assigned.

Reply:
For an application or analytic service which is "discovering" devices already in core-metadata, it could discover based on the  profileName assigned by the south-bound service and then apply its own device profile (perhaps then adding an extended device to core-metadata).

Reply:
The issue I see with this is the current provision watchers operate on a `DiscoveredDevice` object, not the Device object. At least in the Go SDK... 
https://github.com/edgexfoundry/device-sdk-go/blob/main/pkg/models/protocoldiscovery.go#L25-L29

So what you are describing is a second level of provision watchers that operate on the `Device` objects once that have be instantiated in Core Metadata. The is where DSEs come in to detect them, but you still need this new type of Provision Watcher to filter for the ones you care about. If you just need profile name, you could use the MessageBus subscriptions to filter for the ones you care about. 
https://docs.edgexfoundry.org/2.3/microservices/core/metadata/Ch-Metadata/#publish-topic

Reply:
@lenny-intel OK to mark this comment Resolved now?